src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1582,6 +1586,8 @@ private void GatherArrayStringAndValueOutputs(bool outputTargetIsItem, string ou
                             }
                         }
 
+                        PropertyTrackingUtils.LogPropertyAssignment(_propertyTrackingSettings, outputTargetName, outputString, parameterLocation, _projectInstance.GetProperty(outputTargetName)?.EvaluatedValue ?? null, _taskLoggingContext);
```

Comment:
note that we will call into `GetProperty()` even if tracking is turned off completely. This may result in paying the perf cost even if it's not needed. Perhaps surround in an if statement.